platform-sdk/swirlds-merkledb/src/main/java/com/swirlds/merkledb/files/hashmap/HalfDiskHashMap.java

```diff
@@ -855,6 +888,42 @@ private Bucket readBucket(final int bucketIndex) throws IOException {
         return bucket;
     }
 
+    // -- Resize --
+
+    /**
+     * Check if this map should be resized, given the new virtual map size. If the new map size
+     * exceeds 80% of the current number of buckets times {@link #goodAverageBucketEntryCount},
+     * the map is resized by doubling the number of buckets.
+     *
+     * @param firstLeafPath The first leaf virtual path
+     * @param lastLeafPath The last leaf virtual path
+     */
+    public void resizeIfNeeded(final long firstLeafPath, final long lastLeafPath) {
+        final long currentSize = lastLeafPath - firstLeafPath + 1;
+        if (currentSize / numOfBuckets.get() <= goodAverageBucketEntryCount * 80 / 100) {
```

Comment:
This formula should be discussed, I'm open to any feedback. Here is how it's currently coded. Assume VM size hint was originally 1M. With default configs, it would result in 32K buckets created. Once VM actually grows to 80% of the original size hint, that is to 800K, HDHM will double the number of buckets to 64K. When VM grows to 1600K, the number of buckets will grow to 128K. And so on

Reply:
It's better to avoid integer division:
```suggestion
        private final long goodAverageBucketEntryCount;
        
        if (currentSize * 100 <= goodAverageBucketEntryCount * numOfBuckets.get() * 80) {
```
I ran a numerical experiment with uniform key hash code distribution: at 80% load, approximately 9% buckets exceed 32 entries (`goodAverageBucketEntryCount`). At 75% load - 4.6% and at 70% - 2%.
I think 70% provides a better trade-off.